packages/frontend/src/components/Dialog/Dialog.tsx

```diff
@@ -29,13 +27,52 @@ const Dialog = React.memo<Props>(
     height,
     ...props
   }) => {
+    canEscapeKeyClose
+
+    const dialog = useRef<HTMLDialogElement>(null)
+
+    const onClick = canOutsideClickClose
+      ? (ev: React.MouseEvent<HTMLDialogElement>) => {
+          if (!dialog.current) {
+            return
+          }
+          ev.stopPropagation()
+          // that is the way to check if we clicked on dialog::backdrop
+          const rect = dialog.current.getBoundingClientRect()
+          const isInDialog =
+            rect.top <= ev.clientY &&
+            ev.clientY <= rect.top + rect.height &&
+            rect.left <= ev.clientX &&
+            ev.clientX <= rect.left + rect.width
+          if (!isInDialog) {
+            dialog.current.close()
+          }
+        }
+      : () => {}
+
+    const onClose = () => {
+      dialog.current!.style.display = 'none'
+      props.onClose()
+    }
+
+    const onCancel = (ev: React.BaseSyntheticEvent) => {
+      if (!canEscapeKeyClose) {
+        ev.preventDefault()
+      }
+    }
+
+    useEffect(() => {
+      // calling showModal is "only" the way to have ::backdrop
+      dialog.current?.showModal()
+      dialog.current!.style.display = 'flex'
```

Comment:
```suggestion
      dialog.current!.style.display = ''
```

So that CSS styles take over.

Reply:
It's been a while I'll look it up again, but I believe there were some forms which didn't play nicely with the default 'block' display, I'll recheck that. And my bad for not leaving a comment here.

Reply:
What I'm saying is that `flex` should be defined statically in CSS, and when the dialog needs to be hidden, let's apply `style.display = 'none'`.